### ISSUE TITLE
Table sync problem in mixed versions: provider 2.2.2 and subscriber 2.3.0

### ISSUE THREAD

**Issue by ZiBiS** [2ndQuadrant/pglogical#249]
After upgrade pglogical extension on subscriber node replication worked properly, till adding the new table to the existing replication_set on provider node.
Synchronize table data process still stared with errors:

```
ERROR:  schema "'public'" does not exist at character 161
SELECT i.relid, i.nspname, i.relname, i.att_list, i.has_row_filter, i.nspname as nsptarget, i.relname as reltarget  FROM pglogical.show_repset_table_info('''public''.''my_new_table'''::regclass, ARRAY['my_replication_set']) i
```

**Comment by bbellrose1** [comment]
I had a similar issue. Provider node is version 2.2.0 and subscriber is 2.3.0. Received [pglogical apply 30692:2355714646] [[unknown]] ERROR:  could not get table list: ERROR:  syntax error at or near "."

Does not look like the 2.3 version will compile on my source server though. Is pglogical not backward compatible?

**Comment by diranged** [comment]
@ZiBiS:
  I ran into the _same problem_, _last night_,. We were trying to migrate from Postgres 9.6 w/ PGLogical 2.3.0 to RDS Postgres 10.6 w/ PGLogical 2.2.0. We've done this a few times before with the older PGLogical 2.2x series. We failed 5 migrations in a row last night at different random points in the data copy process. The messages we get in the receiver-end logs are pretty terse:

```2020-03-12 18:31:51 UTC::@:[7797]:DEBUG: registering background worker "pglogical apply 16396:2875150205"
2020-03-12 18:31:51 UTC::@:[7797]:DEBUG: starting background worker process "pglogical apply 16396:2875150205"
2020-03-12 18:31:51 UTC::@:[7797]:DEBUG: unregistering background worker "pglogical apply 16396:2875150205"
2020-03-12 18:31:51 UTC::@:[7797]:LOG: worker process: pglogical apply 16396:2875150205 (PID 5870) exited with exit code 1
```

However, I tried downgrading to 2.2.1 on the sender-side an I ran into the same problem. :(

**Comment by Mysticwaver** [comment]
I've had a following issue. We use pglogical 2.2.2 as an extension for provider node. During preparing second subscriber node, default version from ubuntu apt repository was 2.3.0. Almost all steps went good, until we tried synchronize subscription with `pglogical.alter_subscription_synchronize` . There was an error from above. Can we count on backward compatibility fix between 2.3.0 and 2.2.2? Will the only possible way remain updating the version on the provider node, or downgrade the subscriber one?

**Comment by petere** [comment]
This has been fixed in version 2.3.1.

**Comment by bbellrose1** [comment]
No worries, I abandoned my attempts at using pglogical anyway. Does not seem like it was quite ready so going to use just basic postgresql tools to perform the work.
